packages/cli/src/util/deploy/get-deployment-by-id-or-url.ts

```diff
@@ -0,0 +1,89 @@
+import chalk from 'chalk';
+import getDeployment from '../get-deployment';
+import getTeamById from '../teams/get-team-by-id';
+import { isValidName } from '../is-valid-name';
+import type Client from '../client';
+import type { Deployment, Team } from '@vercel-internals/types';
+
+/**
+ * Renders feedback while retrieving a deployment, then validates the
+ * deployment belongs to the current team.
+ *
+ * @param client - The CLI client instance.
+ * @param contextName - The context/team name.
+ * @param deployIdOrUrl - The deployment id or URL.
+ * @returns The deployment info.
+ */
+export async function getDeploymentByIdOrURL({
+  client,
+  contextName,
+  deployIdOrUrl,
+}: {
+  client: Client;
+  contextName: string;
+  deployIdOrUrl: string;
+}): Promise<Deployment> {
+  const { config, output } = client;
+
+  if (!isValidName(deployIdOrUrl)) {
+    throw new Error(
+      `The provided argument "${deployIdOrUrl}" is not a valid deployment or project`
```

Comment:
<!-- Suggestion comments propose improvements to the code. Mention specifically what should be changed and what the benefit would be.  -->
**suggestion (soft-blocking):** don't mention project

```suggestion
      `The provided argument "${deployIdOrUrl}" is not a valid deployment ID or URL`
```